main.rb

```diff
@@ -0,0 +1,197 @@
+# Q1
+students = [
+  {
+      first_name: 'Ahmed',
+      last_name: 'Al-Bahrani'
+  },
+  {
+      first_name: 'Reem',
+      last_name: 'AlHarbi',
+  },
+  {
+      first_name: 'Mansour',
+      last_name: 'Almohsen',
+  }
+]
+
+arr=students.map do |h|
+h.map do |_key,val|
+ upper=val.upcase
+end
+end
+index=0
+arr.map do |a|
+
+
+puts upper_case_full_names = arr[index].join(" ")
+index +=1
+end
+
+#Q2
+users = [
+  {
+      name: 'Hanan',
+      orders: [
+          {
+              description: 'a bike'
+          }
+      ]
+  },
+  {
+      name: 'Abdulrahman',
+      orders: [
+          {
+              description: 'bees'
+          },
+          {
+              description: 'fishing rod'
+          }
+      ]
+  },
+  {
+      name: 'Turki',
+      orders: [
+          {
+              description: 'a MacBook'
+          },
+          {
+              description: 'The West Wing DVDs'
+          },
+          {
+              description: 'headphones'
+          },
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+3.times { |i| puts users[i][:orders][0] }
+
+# Q3
+
+people = [
+  {
+      name: 'Ahlam',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: 'Sulaiman',
+      transactions: [
+          {
+              type: 'BIKES',
+              amount: 800.00
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: 'Norah',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'COFFEE',
+              amount: 100.00
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  }
+]
+people.map do |arr|
+
+    total=[]
+    arr[:transactions].map do |arr2|
+    if arr2[:type] =="COFFEE"
+     total.push(arr2[:amount])
+    
+    end
+    end
+    
+    p name = arr[:name]
+    new_total = total.reduce(:+)
+    p the_average =new_total/ total.length
+    
+    end
+# I locked up this code after I tried the problem for an hour
```

Comment:
Do you understand it? Do you think you can solve a similar problem on your own in the future now that you've seen how it can be approached?